### ISSUE TITLE
Fatal error in PHP version 7.x.x

### ISSUE THREAD

**Issue by NabiKAZ** [sstur/js2php#6]
example.js
```
var x = "world";
console.log("hi " + x);
```

```
>js2php example.js > example.php
Adding runtime file: php/helpers/operators.php
Adding runtime file: php/helpers/helpers.php
Adding runtime file: php/classes/Object.php
Adding runtime file: php/classes/Function.php
Adding runtime file: php/classes/Global.php
Adding runtime file: php/classes/Array.php
Adding runtime file: php/classes/Date.php
Adding runtime file: php/classes/RegExp.php
Adding runtime file: php/classes/String.php
Adding runtime file: php/classes/Number.php
Adding runtime file: php/classes/Boolean.php
Adding runtime file: php/classes/Error.php
Adding runtime file: php/classes/Exception.php
Adding runtime file: php/classes/Buffer.php
Adding runtime file: php/globals/globals.php
Adding runtime file: php/globals/Math.php
Adding runtime file: php/globals/JSON.php
Adding runtime file: php/globals/console.php
Adding runtime file: php/globals/process.php
Processing file `example.js` ...
Output sent to STDOUT
```

example.php
```
:
:
$x = null;
$x = "world";
call_method($console, "log", _concat("hi ", $x));
```

**Test**

- [x] PHP 5.3.13
```
hi world
```

- [x] PHP 5.5.12
```
hi world
```

- [x] PHP 5.6.35
```
hi world
```

- [ ] PHP 7.0.29
```
Fatal error: Cannot use 'Null' as class name as it is reserved in example.php on line 575
```

- [ ] PHP 7.1.16
```
Fatal error: Cannot use 'Null' as class name as it is reserved in example.php on line 575
```

- [ ] PHP 7.2.4
```
Fatal error: Cannot use 'Object' as class name as it is reserved in example.php on line 312
```

Some parts of code example.php
```
:
:
312   class Object {
313   public $data = null;
314   public $proto = null;
:
:
575   class Null {}
576   Object::$null = new Null();
577   Object::$protoObject = new Object();
578   Object::$protoObject->proto = Object::$null;
:
:
```

**Comment by cptolemy** [comment]
Hi NabiKAZ

I have the same issue. Were you able to fix it? Can you share your solution if that's the case?

Kind regards,

**Comment by NabiKAZ** [comment]
@cptolemy No, seem this is bug in core. I use [phantomjs ](https://github.com/ariya/phantomjs) and [php-phantomjs](https://github.com/jonnnnyw/php-phantomjs/) for execute js and render page.

**Comment by sstur** [comment]
Thanks for the detailed issue @NabiKAZ!

It looks like PHP 7+ has added new reserved words. Seems those classes would need to be renamed.

This project is pretty old and I don't really maintain it but if someone wants to make a PR, I'll accept it.

Thanks!

**Comment by cptolemy** [comment]
Greetings @NabiKAZ and @sstur 

Here's a simple report of the major bugs I've found in my sample file:

--------------------------------------------------------------------------------
Found 14 spot(s), 4 identified
--------------------------------------------------------------------------------
   12 | NOTICE     |   | 5.3.0 | call_user_func_array() no longer accept non-array passed as arguments
  278 | NOTICE     |   | 7.0.0 | Modulus operator will throw a exception if divisor is 0
  312 | NOTICE     | * | 7.0.0 | Keyword "Object" is reserved
  458 | NOTICE     |   | 5.3.0 | call_user_func_array() no longer accept non-array passed as arguments
  575 | FATAL      | * | 7.0.0 | Keyword "Null" cannot be used to name class-like
  645 | NOTICE     |   | 5.3.0 | call_user_func_array() no longer accept non-array passed as arguments
 1040 | NOTICE     |   | 5.3.0 | call_user_func_array() no longer accept non-array passed as arguments
 1053 | NOTICE     |   | 5.3.0 | call_user_func_array() no longer accept non-array passed as arguments
 1133 | NOTICE     |   | 5.3.0 | uasort() no longer accept objects passed as arguments
 1637 | DEPRECATED |   | 5.5.0 | preg_replace() /e modifier is deprecated, use preg_replace_callback() instead
 1837 | FATAL      | * | 7.0.0 | Cannot redeclare class "Error"
 1883 | FATAL      | * | 7.0.0 | Cannot redeclare class "TypeError"
 2007 | WARNING    |   | 7.0.0 | set_exception_handler() is no longer guaranteed to receive Exception objects
 2445 | NOTICE     |   | 5.6.0 | json_decode() rejects non-lowercase variants of true, false, null
--------------------------------------------------------------------------------

I'm working on the substitution of the class names. Do you want to join me?

Kind regards,

**Comment by cptolemy** [comment]
Good evening,

With some minor changes, I was able to run my code without problems. PHP v7 does bring some annoying changes.

So the Null, Error and TypeError errors are fixed - I'll make some other tests regarding other problems, like deprecated functions or other possible class problems.

The zip file includes the php folder (with 4 sub folders) of jstophp: [php.zip](https://github.com/sstur/js2php/files/2131286/php.zip) 

Kind regards,

CPtolemy

**Comment by zaoqi** [comment]
https://github.com/sstur/js2php/commit/5fad48e075e257da34ba8af88424174087e4ac70

Is it fixed?
